protocol/src/main/java/io/camunda/zeebe/protocol/Protocol.java

```diff
@@ -68,6 +68,12 @@
   /** Task header key used for user tasks to contain form key from BPMN XML */
   public static final String USER_TASK_FORM_KEY_HEADER_NAME = "io.camunda.zeebe:formKey";
 
+  /** Task header key used for assignee */
+  public static final String TASK_ASSIGNEE_HEADER_NAME = "io.camunda.zeebe:assignee";
+
+  /** Task header key used for candidate groups */
+  public static final String TASK_CANDIDATE_GROUPS_HEADER_NAME = "io.camunda.zeebe:candidateGroups";
```

Comment:
:wrench: We could align the names with the other fields. The other user task fields have the prefix `USER_TASK_` instead of just `TASK_`. 

```suggestion
  public static final String USER_TASK_ASSIGNEE_HEADER_NAME = "io.camunda.zeebe:assignee";

  /** Task header key used for candidate groups */
  public static final String USER_TASK_CANDIDATE_GROUPS_HEADER_NAME = "io.camunda.zeebe:candidateGroups";
```

Reply:
Interesting, but its a trade off with flexibility. If we don't focus it to user tasks specifically we might be able to apply task assignments to other tasks in the future. Consider for example assigning jobs to a specific job worker instead of only to a type (for which multiple job workers can exist). 

I understand we don't have this now, and we may never but currently the implementation is allowing to quickly implement it. Do we want to restrict that? What is your opinion?

Reply:
Interesting point :bulb: 

I prefer to restrict the task header. For me, it's very specific to user tasks. If we implement something like this for service tasks/jobs then we could add something specifically for jobs.